packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1920,7 +1920,9 @@ class _OverlayPortalState extends State<OverlayPortal> {
 
   void _setupController(OverlayPortalController controller) {
     assert(
-      controller._attachTarget == null || controller._attachTarget == this,
+      controller._attachTarget == null ||
```

Comment:
nit: looks like the first check is also covered by the 3rd check? If so can we remove the first line?

Reply:
Sure, Let me push the change.